### ISSUE TITLE
API/BUG: Index creation with named Index resets name

### ISSUE THREAD

**Issue by sinhrks** [pydata/pandas#11193]
```
import pandas as pd
idx = pd.Index([1, 2 ,3], name='x')
pd.Index(idx)
# Int64Index([1, 2, 3], dtype='int64')
```

A point is whether to reset existing Index name in below case. Current impl can't distinguish whether name is not passed, or passed ``None`` explicitly.

```
pd.Index(idx, name=None)
# -> should reset name?
```

**Comment by jreback** [comment]
Same for ``DatetimeIndex``. smells like a buggy

```
In [18]: pd.DatetimeIndex(pd.date_range('20130101',periods=3,name='foo'))
Out[18]: DatetimeIndex(['2013-01-01', '2013-01-02', '2013-01-03'], dtype='datetime64[ns]', freq='D')
```

**Comment by mbirdi** [comment]
I tried to reproduce the bug and failed.   Here is a link the code examples I ran on my laptop.

http://nbviewer.ipython.org/gist/mbirdi/7586ebc290bd90a73ed3

So in my examples I was able to reset the name of the index from 'x' to None.

**Comment by jreback** [comment]
@mbirdi this reproduces on master

```
In [58]: idx = pd.Index([1, 2 ,3], name='x')

In [59]: pd.Index(idx)
Out[59]: Int64Index([1, 2, 3], dtype='int64')

In [60]: pd.Index(idx).name
```

**Comment by mbirdi** [comment]
@jreback
@sinhrks  

Thanks for the feedback.  

I see the question up for discussion now.  I have never really had to create an index from an index.  Is that a common thing to do in pandas?  I am trying to picture why I would do that with the datasets I work with and I am coming up blank.

**Comment by jreback** [comment]
well if I have something that could be a list or an Index then I want to be able to wrap it and preserve meta data. The bottom line is that we want to have API consistency.

**Comment by IamGianluca** [comment]
I can work on this. I'll send a PR shortly.

**Comment by IamGianluca** [comment]
@jreback What should be the desired behaviour when the user passes `None` as the value for the argument `name` and the existing index we are passing had already a name?

```
idx = pd.Index([1, 2 ,3], name='x')
pd.Index(idx, name=None)
# -> should reset name?
```

**Comment by shoyer** [comment]
Well, one option is to use a sentinel object for the default instead, e.g., `_default = object()`, and then use `def Index(..., name=_object):`. Then you can tell the difference between the user passing `name=None` and not passing it at all. That said, I don't know if this is worth worrying about it. It's probably OK to keep the current name if `name=None` is passed.

**Comment by jreback** [comment]
I agree with @shoyer here. The point of this is to preserve the name, so ``name=None`` is not passing one (yes could do something like @shoyer suggest, but not worth it IMHO).

**Comment by IamGianluca** [comment]
Thanks for the clarification @jreback and @shoyer . I will send a pull request this evening.

**Comment by sinhrks** [comment]
The last question (`name=None`) is derived from test failures when I tried to fix it by simply retrieving the given `Index` name. @shoyer 's suggestion should work and be consistent.

**Comment by jreback** [comment]
closed by #11695
